aries_cloudagent/connections/base_manager.py

```diff
@@ -285,8 +295,9 @@ async def fetch_connection_targets(
             self._logger.debug("No local DID associated with connection")
             return None
 
-        wallet = self._session.inject(BaseWallet)
-        my_info = await wallet.get_local_did(connection.my_did)
+        async with self._profile.session() as session:
```

Comment:
I think this could be split similarly as you did in line 170. It's not used in all cases.

Reply:
Nice catch. Moved tnx